docs/docs/code/bash/README.md

```diff
@@ -0,0 +1,199 @@
+
+### Making a GET request
+
+You can use `curl` to perform GET requests from websites or APIs directly.
+
+```bash
+# Get the current weather in San Francisco
+WEATHER=`curl --silent https://wttr.in/San\ Francisco\?format=3`
+
+echo $WEATHER
+# Produces:
+# San Francisco: 🌫  +48°F
+```
+
+::: tip
+Use the `--silent` flag with `curl` to suppress extra extra diagnostic information that `curl` produces when making requests.
+
+This enables you to only worry about the body of the response so you can visualize it with tools like `echo` or `jq`. 
+:::
+
+### Making a POST request
+
+`curl` handles making POSTs requests as well. The `-X` flag allow you to specify the HTTP method you'd like to use for an HTTP request.
+
+The `-d` flag is for passing data in the POST request.
+
+```bash
+curl --silent -X POST https://postman-echo.com/post -d 'name=Bulbasaur&id=1'
+
+# to store the API response in a variable, interpolate the response into a string and store it in variable
+RESPONSE=`curl --silent -X POST https://postman-echo.com/post -d 'name=Bulbasaur&id=1'`
+
+# now the response is stored as a variable
+echo $RESPONSE
+```
+
+### Using API key authentication
+
+Some APIs require you to use a secret API key to prove your access.
+
+`curl` has an `-h` flag where you can pass your API key as a token.
+
+Below is an example of searching Twitter for mentions of a specific Twitter handle. Twitter requires a valid API key passed in the header of the request.
+
+```bash
+curl --silent -X POST -h "authorization:Bearer $(<your api key here>)" https://api.twitter.com/2/users/@pipedream/mentions
+```
+
+## Sharing data between steps
+
+A step can accept data from other steps in the same workflow, or pass data downstream to others.
+
+This makes your steps even more powerful, you can compose new workflows and reuse steps.
+
+### Using data from another step
+
+In bash scripts, data from the initial workflow trigger and other steps are available in the `$PIPEDREAM_STEPS` environment variable.
+
+In this example, we'll pretend this data is coming into our HTTP trigger via a POST request.
+
+```json
+{
+  "id": 1,
+  "name": 'Bulbasaur",
+  "type": "plant"
+}
+```
+
+In our bash script, we can access this data via the `$PIPEDREAM_STEPS` variable. Specifically, this data from the POST request into our workflow is available in the `trigger` object.
+
+```bash
+echo $PIPEDREAM_STEPS | jq .trigger.event
+
+# results in { id: 1, name: "Bulbasaur", type: "plant" }
+```
+
+::: tip
+The period (`.`) in front the `trigger.event` in the example is not a typo. This is to define the starting point for `jq` to start traversing down the JSON in the HTTP response.
+:::
+
+### Sending data downstream to other steps
+
+To share data created, retrieved, transformed or manipulated by a step to others downstream, append it to the `$PIPEDREAM_EXPORTS` variable.
```

Comment:
"To share data created, retrieved, transformed or manipulated by a step to others downstream" -> "To return data for future steps to use"

Reply:
Like `$PIPEDREAM_STEPS` above, `$PIPEDREAM_EXPORTS` now points to a file, so you'll also need to update this section